### PR TITLE
[DOC] corrected Discord channel mention in developer guide

### PR DESCRIPTION
**Pull request by shankariraja**
Fixes: #6162

Changed the channel reference from #contributors to #lobby in the sktime documentation to accurately reflect the Discord server setup. This ensures that new developers receive correct instructions on how to introduce themselves and engage with the community.

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
Thanks! Well-spotted, and apologies for the mistake.

**Review by fkiraly — CHANGES_REQUESTED** [review]
yes, I think `#dev-chat` gives a nicer onboarding experience. `#lobby` also makes sense though. Up to you - which channel did you find more helpful?

**Comment by shankariraja** [comment]
@fkiraly Thanks for your input! I'll make the change soon.

**Review by fkiraly — APPROVED** [review]
Thanks.

Just to check, did you think that's the better channel to direct new contributors to?
It's a bit hard for me to judge from the "inside" perspective, while you have the new contributor perspective.

**Comment by shankariraja** [comment]
From my perspective as a new contributor, having a separate channel dedicated to contributor introductions, like #intro, could provide a more focused and welcoming environment for newcomers. However, I've gone ahead and made the change to #dev-chat as per your suggestion. We can always revisit this decision later based on community feedback. Thanks for being open to it!

**Comment by shankariraja** [comment]
Thank you for creating the #intro channel! I appreciate your responsiveness. I'll update the documentation soon to reflect this change. Thanks again for your help!

**Comment by fkiraly** [comment]
Oh, yes, I wanted to respond here, but then got distracted by discord server settings and how to set up introduction tasks with channel introductions.

Yes, it would be great - if you think that's the right idea to try - to point to `#intro`.

Further, there is another reference on the "get started" page, it's in `contributing.rst` (the file behind this page: https://www.sktime.net/en/latest/get_involved/contributing.html)

I suppose that should also point to `#intro` now?

**Comment by shankariraja** [comment]
Thanks for the update!  I'll make sure to update the "get started" page in contributing.rst to reflect this change. Appreciate your input and efforts in improving the onboarding experience for new contributors!

File: docs/source/get_involved/contributing.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ and people who are looking to learn and develop their skills.
 
 Recommended steps for first time contributors, or to get started with regular contributions:
 
-1. Say hello (in the lobby on `Discord`_)!
+1. Say hello (in the ``#intro`` on `Discord`_)!
```
Comment (fkiraly) [inline docs/source/get_involved/contributing.rst:27]: intro channel, no? word "channel" missing

Comment (shankariraja) [reply docs/source/get_involved/contributing.rst:27]: Sure, I'll change

**Review by fkiraly — APPROVED** [review]

**Pull request merged**